### PR TITLE
51 react create the search page and include its components

### PR DESCRIPTION
**Pull request by naghamabuwarda**

### CODE REVIEW

File: client/src/components/filter/index.tsx [hunk 1]
```diff
@@ -10,29 +11,33 @@ interface Props {
...
           />
-          <label id="demo-radio-buttons-group-label" htmlFor="category" className="label catelabel">Categories</label>
+          <label id="demo-radio-buttons-group-label" htmlFor="category" className="label catelabel">
+            Categories
+            <CloseIcon onClick={iconChange} />
+          </label>
           <RadioGroup
             aria-labelledby="demo-radio-buttons-group-label"
-            defaultValue={location.state?.category || category}
+            value={location.state?.category || category}
```
Comment (LinaYahya) [inline client/src/components/filter/index.tsx:40]:
```suggestion
            value={category}
            defaultValue={location.state?.category}```

File: client/src/components/filter/index.tsx [hunk 2]
```diff
@@ -10,29 +11,33 @@ interface Props {
...
-  category, changeCategory, priceChange, price,
+  category, changeCategory, priceChange, price, iconChange,
 }: Props) {
   const location = useLocation();
   return (
-    <Container className="content">
+    <Container className="filterContent">
       <div className="sideBar">
         <FormControl>
           <label htmlFor="price" className="label">Price</label>
```
Comment (LinaYahya) [inline client/src/components/filter/index.tsx:26]: change label to Minimum budget, It'll be more descriptive

File: client/src/components/tab/index.tsx [hunk 3]
```diff
@@ -4,41 +4,55 @@ import Tab from '@mui/material/Tab';
 import TabContext from '@mui/lab/TabContext';
 import TabList from '@mui/lab/TabList';
 import TabPanel from '@mui/lab/TabPanel';
+import { TextField } from '@mui/material';
 import './style.css';
 
 interface TabListInt {
   label: string;
-  child: JSX.Element;
+  child: JSX.Element[];
```
Comment (LinaYahya) [inline client/src/components/tab/index.tsx:12]:
```suggestion
 child: JSX.Element | JSX.Element[];
;
```

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: client/src/pages/JobsSearch/index.tsx [hunk 4]
```diff
@@ -1,47 +1,144 @@
...
 function JobsSearch() {
-  const [price, setPrice] = useState<number>(0);
+  // states
+  const [error, setError] = useState(true);
+  const [loading, setLoading] = useState(false);
+  const [jobsCount, setJobsCount] = useState(0);
+  const [page, setPage] = useState<number>(1);
+  const [jobs, setJobs] = useState<Job[]>([]); //
+  const [budget, setPrice] = useState<number>(0);
   const [category, setCategory] = useState<string>('');
```
Comment (LinaYahya) [inline client/src/pages/JobsSearch/index.tsx:36]: set the initial state as `(state?.category` || '')`

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by shathakh — APPROVED** [review]

**Review by LinaYahya — APPROVED** [review]

**Review by mohammedagha27 — APPROVED** [review]

**Pull request merged**